modules/keepnick.cpp

```diff
@@ -174,7 +174,7 @@ class CKeepNickMod : public CModule {
                 return HALT;
             }
 
-            PutModule("Unable to obtain nick: " + numeric.GetParam(2) + ": " + numeric.GetParam(3));
+            PutModule("Unable to obtain nick: " + numeric.GetParam(1) + ": " + numeric.GetParam(2));
```

Comment:
The GetParam calls were definitely correct, although I can see where you were coming from.

By changing GetParam's ints around, the string changed from the following:

```
Unable to obtain nick: ####z-test: Cannot change nickname while banned on channel
Unable to obtain nick: #channel: ReasonByIrcd
```

To:

```
Unable to obtain nick: zarthue: ####z-test
Unable to obtain nick: NickInQuestion: TargetChannel
```

The idea was to include the ircd provided reason so they know *why* the nick change was not going through. I omitted the nick (Param 1) because your primary nickname is unlikely to change, and you are likely to know it already. Inclusion is not a bad thing (Changing the string to "Unable to obtain nick $nick: $channel: $reason"), but I personally omitted it because I did not think it was relevant.